File: SPCode/CS/Modules/RegSupport.cs
using System.Text.RegularExpressions;
using System.Diagnostics;
using System;
using System.Drawing.Drawing2D;
using System.Windows.Forms;
using System.Collections;
using System.Drawing;
using Microsoft.VisualBasic;
using System.Data.SqlClient;
using System.Data;
using System.Collections.Generic;
using WeifenLuo.WinFormsUI;
using Microsoft.Win32;
using WeifenLuo;



namespace SoftLogik.Win
{
	sealed class RegSupport
	{
		
		private static CSPRegistry m_objReg;
		
		public static CSPRegistry Reg
		{
			get
			{
				if (m_objReg == null)
				{
					m_objReg = new CSPRegistry();
				}
				return m_objReg;
			}
		}
		
	}
	
	public class CSPRegistry
	{
		
		#region Public Procedures
		public bool SetValue(ref Microsoft.Win32.RegistryHive KeyRoot, string KeyPath, string KeyName, object Value, string RemoteMachine)
		{
			RegistryKey objKey;
			
			try
			{
				objKey = OpenKey(KeyRoot, KeyName, RemoteMachine, Constants.vbNullString);
				objKey.SetValue(KeyName, Value);
				objKey.Close();
			}
			catch (Exception)
			{
				
			}
			
			return true;
		}
		public object GetValue(Microsoft.Win32.RegistryHive Key, string KeyPath, string Name, object DefaultValue, string RemoteMachine)
		{
			RegistryKey objKey;
			object objValue = null;
			
			try
			{
				switch (Key)
				{
					case RegistryHive.ClassesRoot:
						objKey = Registry.ClassesRoot.OpenSubKey(KeyPath, true);
						break;
					case RegistryHive.LocalMachine:
						objKey = Registry.LocalMachine.OpenSubKey(KeyPath, true);
						break;
					case RegistryHive.CurrentUser:
						objKey = Registry.CurrentUser.OpenSubKey(KeyPath, true);
						break;
					default:
						objKey = RegistryKey.OpenRemoteBaseKey(Key, RemoteMachine).OpenSubKey(KeyPath, true);
						break;
				}
				
				objKey.GetValue(Name);
				objKey.Close();
			}
			catch (Exception)
			{
				
			}
			
			return System.Convert.ToString(objValue);
		}
		
		public bool KeyExists(ref RegistryHive KeyRoot, string KeyName, string RemoteMachine)
		{
			
			return !(OpenKey(KeyRoot, KeyName, RemoteMachine, Constants.vbNullString) == null);
		}
		public bool ValueExists(ref RegistryHive KeyRoot, string KeyName, string ValueName, string RemoteMachine)
		{
			RegistryKey objKey = OpenKey(KeyRoot, KeyName, RemoteMachine, Constants.vbNullString);
			
			return (objKey.GetValue(ValueName) == null);
		}
		#endregion
		
		#region Private Procedures
		private RegistryKey OpenKey(Microsoft.Win32.RegistryHive Key, string KeyPath, string Name, string RemoteMachine)
		{
			
			RegistryKey objKey;
			
			try
			{
				
				switch (Key)
				{
					case RegistryHive.ClassesRoot:
						objKey = Registry.ClassesRoot.OpenSubKey(KeyPath, true);
						break;
					case RegistryHive.LocalMachine:
						objKey = Registry.LocalMachine.OpenSubKey(KeyPath, true);
						break;
					case RegistryHive.CurrentUser:
						objKey = Registry.CurrentUser.OpenSubKey(KeyPath, true);
						break;
					default:
						objKey = RegistryKey.OpenRemoteBaseKey(Key, RemoteMachine).OpenSubKey(KeyPath, true);
						break;
				}
				
				return objKey;
			}
			catch (Exception)
			{
				return null;
			}
			
		}
		#endregion
	}
	
	
}

File: SPCode/Modules/RegSupport.vb

Imports Microsoft.Win32

Module RegSupport

    Private m_objReg As CSPRegistry

    Public ReadOnly Property Reg() As CSPRegistry
        Get
            If IsNothing(m_objReg) Then
                m_objReg = New CSPRegistry
            End If
            Return m_objReg
        End Get
    End Property

End Module

Public Class CSPRegistry
#Region "Public Procedures"
    Public Function SetValue(ByRef KeyRoot As Microsoft.Win32.RegistryHive, _
    ByVal KeyPath As String, ByVal KeyName As String, ByVal Value As Object, _
    Optional ByVal RemoteMachine As String = vbNullString) As Boolean
        Dim objKey As RegistryKey

        Try
            objKey = OpenKey(KeyRoot, KeyName, RemoteMachine)
            objKey.SetValue(KeyName, Value)
            objKey.Close()
        Catch ex As Exception

        End Try

        Return True
    End Function
    Public Function GetValue(ByRef Key As Microsoft.Win32.RegistryHive, _
ByVal KeyPath As String, ByVal Name As String, Optional ByVal DefaultValue As Object = Nothing, _
Optional ByVal RemoteMachine As String = vbNullString) As Object
        Dim objKey As RegistryKey, objValue As Object = Nothing

        Try
            Select Case Key
                Case RegistryHive.ClassesRoot
                    objKey = Registry.ClassesRoot.OpenSubKey(KeyPath, True)
                Case RegistryHive.LocalMachine
                    objKey = Registry.LocalMachine.OpenSubKey(KeyPath, True)
                Case RegistryHive.CurrentUser
                    objKey = Registry.CurrentUser.OpenSubKey(KeyPath, True)
                Case Else
                    objKey = RegistryKey.OpenRemoteBaseKey(Key, _
                    RemoteMachine).OpenSubKey(KeyPath, True)
            End Select

            objKey.GetValue(Name)
            objKey.Close()
        Catch ex As Exception

        End Try

        Return CType(objValue, String)
    End Function

    Public Function KeyExists(ByRef KeyRoot As RegistryHive, _
    ByRef KeyName As String, Optional ByVal RemoteMachine As String = vbNullString) As Boolean

        Return Not (OpenKey(KeyRoot, KeyName, RemoteMachine) Is Nothing)
    End Function
    Public Function ValueExists(ByRef KeyRoot As RegistryHive, _
ByRef KeyName As String, ByVal ValueName As String, _
 Optional ByVal RemoteMachine As String = vbNullString) As Boolean
        Dim objKey As RegistryKey = OpenKey(KeyRoot, KeyName, RemoteMachine)

        Return (objKey.GetValue(ValueName) Is Nothing)
    End Function
#End Region

#Region "Private Procedures"
    Private Function OpenKey(ByRef Key As Microsoft.Win32.RegistryHive, _
    ByVal KeyPath As String, ByVal Name As String, _
    Optional ByVal RemoteMachine As String = vbNullString) As RegistryKey

        Dim objKey As RegistryKey

        Try

            Select Case Key
                Case RegistryHive.ClassesRoot
                    objKey = Registry.ClassesRoot.OpenSubKey(KeyPath, True)
                Case RegistryHive.LocalMachine
                    objKey = Registry.LocalMachine.OpenSubKey(KeyPath, True)
                Case RegistryHive.CurrentUser
                    objKey = Registry.CurrentUser.OpenSubKey(KeyPath, True)
                Case Else
                    objKey = RegistryKey.OpenRemoteBaseKey(Key, _
                    RemoteMachine).OpenSubKey(KeyPath, True)
            End Select

            Return objKey
        Catch ex As Exception
            Return Nothing
        End Try

    End Function
#End Region
End Class

